src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1114,6 +1114,11 @@ public sealed class ValidatePatternAttribute : ValidateEnumeratedArgumentsAttrib
         public RegexOptions Options { set; get; } = RegexOptions.IgnoreCase;
 
         /// <summary>
+        /// Gets or sets the custom error message that is displayed to the user
+        /// </summary>
+        public string ErrorMessage { get; set; }
```

Comment:
I can see wanting a custom message with a few others. ValidateScript if you don't want to throw, and ValidateSet if you have too many possibilities and a shorter message is clearer.